src/common/snippets/src/op/subgraph.cpp

```diff
@@ -462,40 +522,43 @@ snippets::Schedule Subgraph::generate(const BlockedShapeVector& blocked_input_sh
                                       const std::shared_ptr<IShapeInferSnippetsFactory>& factory,
                                       const void* compile_params) {
     data_flow_transformations(blocked_input_shapes, input_precisions, output_precisions, data_flow_backend_passes);
-    convert_body_to_linear_ir(min_parallel_work_amount, min_kernel_work_amount, factory);
-    return generate_from_linear_ir(lowered_pass_config, lowered_backend_passes, compile_params);
+    lowering_transformations(min_parallel_work_amount, min_kernel_work_amount, factory, lowered_pass_config, lowered_backend_passes);
+    return generate_from_linear_ir(compile_params);
 }
 
-snippets::Schedule Subgraph::generate_from_linear_ir(const std::shared_ptr<lowered::pass::PassConfig>& lowered_pass_config,
-                                                     const std::vector<snippets::lowered::pass::PassPipeline::PositionedPassLowered>& backed_passes,
-                                                     const void* compile_params) const {
+snippets::Schedule Subgraph::generate_from_linear_ir(const void* compile_params) const {
     INTERNAL_OP_SCOPE(Subgraph);
     OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::op::generate")
     OPENVINO_ASSERT(m_generator != nullptr, "generate is called while generator is not set");
 
     // actual code emission
-    // Note: some transformations performed in the generator, e.g. tail insertion, can break shape propagation
-    //  until we fix this behavior, we have to make a copy of LIR before giving it to the generator.
+    // Note: to not damage the lowered linear IR for the shape-dependent passes, we have to make a copy
     OPENVINO_ASSERT(m_linear_ir, "Attempt to call generate, when linear IR was not initialized");
-    auto linear_ir {*m_linear_ir->clone()};
-    LoweringResult lowering_result;
-    control_flow_transformations(linear_ir, lowering_result, lowered_pass_config, backed_passes);
-#ifdef SNIPPETS_DEBUG_CAPS
-    if (linear_ir.get_config().perf_count_mode != lowered::PerfCountMode::Disabled) {
-        lowered::pass::InsertPerfCount perf_count_pass({});
-        perf_count_pass.run(linear_ir, linear_ir.cbegin(), linear_ir.cend());
+    auto linear_ir = *lowered::LinearIRBuilder().clone(m_linear_ir);
+
+    if (is_dynamic()) {
+        ov::snippets::lowered::pass::PassPipeline shape_dependent_pipeline;
+        shape_dependent_pipeline.register_pass<ov::snippets::lowered::pass::SetLoadStoreScalar>();
+        shape_dependent_pipeline.register_pass<ov::snippets::lowered::pass::InsertBroadcastMove>();
+        shape_dependent_pipeline.register_pass<ov::snippets::lowered::pass::LoadMoveBroadcastToBroadcastLoad>();
```

Comment:
Since we are trying to minimize runtime overheads in dynaimc scenario,it looks like we should rewrite it in one pass.
Also it seems that we insert Brodacasts only after Loads now, no? In this case we can replace Load with BrodcastLoad straight away.